### PR TITLE
Renaming from OT to OTel

### LINKED ISSUE #1042: Rename "ot" to "otel"
OpenTelemetry specification has officially recommended "otel" as the abbreviation rather than "ot" (which could be confusing as it is used for OpenTracing).

Need to find all `ot` names and update them to `otel` when appropriate: https://github.com/open-telemetry/opentelemetry-dotnet/search?q="ot."&unscoped_q="ot."

Reference: https://github.com/open-telemetry/opentelemetry-specification#acronym

### PR DESCRIPTION
**Pull request by eddynaka**
Fixes #1042.

## Changes

Please provide a brief description of the changes here. Update the
`CHANGELOG.md` for non-trivial changes.

For significant contributions please make sure you have completed the following items:

* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: examples/Console/TestConsoleExporter.cs [hunk 1]
```diff
@@ -41,14 +41,14 @@ internal static object Run(ConsoleOptions options)
             // which decide to use Open Telemetry.
 
             // Libraries would simply write the following lines of code to
-            // emit activities, which are the .NET representation of OT Spans.
+            // emit activities, which are the .NET representation of OTel Spans.
```
Comment (reyang) [inline examples/Console/TestConsoleExporter.cs:44]: For comments, I think we probably want to use the full name given we already use the full name in the copyright comment `// Copyright The OpenTelemetry Authors`.

File: test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusExporterTests.cs [hunk 2]
```diff
@@ -182,7 +182,7 @@ private void ValidateResponse(string responseText)
             // Once a pull model is implemented, we'll not have this issue and we need to add tests
             // at that time.
 
-            // If in future, there is a official .NET Prometheus Client library, and OT Exporter
+            // If in future, there is a official .NET Prometheus Client library, and OTel Exporter
```
Comment (reyang) [inline test/OpenTelemetry.Exporter.Prometheus.Tests/PrometheusExporterTests.cs:185]: a official -> an official

File: examples/Console/TestConsoleExporter.cs (hunk 1 above)
Comment (eddynaka) [reply examples/Console/TestConsoleExporter.cs:44]: OK!

File: test/OpenTelemetry.Instrumentation.Http.Tests/HttpClientTests.netcore31.cs [hunk 3]
```diff
@@ -124,7 +124,7 @@ public async Task HttpOutCallsAreCollectedSuccessfullyAsync(HttpTestData.HttpOut
                 Assert.Equal(tc.SpanStatusHasDescription.Value, !string.IsNullOrEmpty(desc));
             }
 
-            var normalizedAttributes = span.TagObjects.Where(kv => !kv.Key.StartsWith("ot")).ToImmutableSortedDictionary(x => x.Key, x => x.Value.ToString());
+            var normalizedAttributes = span.TagObjects.Where(kv => !kv.Key.StartsWith("otel")).ToImmutableSortedDictionary(x => x.Key, x => x.Value.ToString());
```
Comment (reyang) [inline test/OpenTelemetry.Instrumentation.Http.Tests/HttpClientTests.netcore31.cs:127]:
```suggestion
            var normalizedAttributes = span.TagObjects.Where(kv => !kv.Key.StartsWith("otel.")).ToImmutableSortedDictionary(x => x.Key, x => x.Value.ToString());
```

**Comment by eddynaka** [comment]
> LGTM.

@reyang , just updated everything. Can you re-review?

File: src/OpenTelemetry.Api/Trace/ActivityExtensions.cs [hunk 4]
```diff
@@ -30,7 +30,7 @@ public static class ActivityExtensions
         /// <summary>
         /// Sets the status of activity execution.
         /// Activity class in .NET does not support 'Status'.
-        /// This extension provides a workaround to store Status as special tags with key name of ot.status_code and ot.status_description.
+        /// This extension provides a workaround to store Status as special tags with key name of otel.status_code and otel.status_description.
```
Comment (cijothomas) [inline src/OpenTelemetry.Api/Trace/ActivityExtensions.cs:33]: can you update changelog for this change?

Comment (eddynaka) [reply src/OpenTelemetry.Api/Trace/ActivityExtensions.cs:33]: just updated the api/changelog.

**Pull request merged**